### ISSUE TITLE
Why we should avoid d_loss converge to zero?

### ISSUE THREAD

**Issue by Captainsean007** [khalooei/ALOCC-CVPR2018#12]
when I reproduct this code on my own application, I can easily get d_loss with 0 and g_loss with 20. 
![image](https://user-images.githubusercontent.com/34062397/55930354-6bac3680-5c53-11e9-951d-9a0eee77a261.png) How could I fix this up, your any advices would be helpful, thinks!

**Comment by khalooei** [comment]
Dear @Captainsean007 , It depends on many things. Please check and tune your hyper parameters, check each normalization step which you select in your dataset.  Also, you can select a lower learning rate  to move more precisely in your learning manifold when the ALOCC tries to do the optimization phases.

**Comment by Captainsean007** [comment]
Dear @khalooei , thanks for your answers. I didn't do any pre-processing before feed data to graph(just resize it with the shape of (224,224,3)). Noise_data which is fed into G net is produced by input data add some gassain noises, I wonder that if it's important with the variance of Gassain noises of different dataset? should I change this variance you used in mnist dataset. Thanks for your help again sincerely!

**Comment by khalooei** [comment]
Your welcome! Please do the data normalization step before feeding any things to neural networks. Therefore, in our case, please do it in the beginning steps of your dataset feeding and then feed them to the model. If you need any help, don't hesitate to contact me.

**Comment by Captainsean007** [comment]
OK, a optimization phases problem. When you trained the network, you optimized G network twice to avoid d_loss equal to zero. Why is this work? And what is the network learned when d_loss is 0.

**Comment by khalooei** [comment]
As we mentioned in our implementation and also paper, the optimization phases of this implementation are inherited from Deep convolution Generative Adversarial Network (DCGAN) implementation. When the d_loss is equal to zero, it shows that the D network overcomes G and the game tries to be finished with one winner and this is not good for learning the framework of GANs because we must try to implement the game theory problem in this problem. At this moment, If I remember, we inherit it from DCGAN which they mentioned that it could implement the game again better.  You can read the Generative Adversarial Network tutorial of Goodfellow ( @goodfeli ) in NIPS 2016 (https://arxiv.org/pdf/1701.00160.pdf) to learn more information about this approach. Also, you can follow the DCGAN implementation (https://carpedm20.github.io/faces) for more information about stabilizing the deep convolutional version of GANs. If you need any help, don't hesitate to contact me.

**Comment by Captainsean007** [comment]
OK, thanks. Have a good day！

**Comment by khalooei** [comment]
You're welcome. Have a good day too!
